data/hydraulic-data.py

```diff
@@ -0,0 +1,129 @@
+"""Create hydraulic systems dataset"""
+
+from typing import Union, List
+from h2oaicore.data import CustomData
+import datatable as dt
+import numpy as np
+import pandas as pd
+
+class HydraulicSystemsData(CustomData):
+    # url contains hydraulic systems condition monitoring data set
+    base_url = "https://archive.ics.uci.edu/ml/machine-learning-databases/00447/"
+
+    @staticmethod
+    def create_data(X: dt.Frame = None) -> Union[str, List[str],
+                                                 dt.Frame, List[dt.Frame],
+                                                 np.ndarray, List[np.ndarray],
+                                                 pd.DataFrame, List[pd.DataFrame]]:
+        import os
+        from h2oaicore.systemutils_more import download
+        from h2oaicore.systemutils import config
+        import zipfile
+
+        def mean_conversion(pd_frame):
```

Comment:
Just leaving this here for reference.  Type hints may be helpful, not just for type checkers/linters/etc., but also improve readability by specifying what the expected argument and return types are.

https://docs.python.org/3/library/typing.html

Reply:
Thank you for the suggestion on using Type hints for improving readability. For each of my relevant methods in the class HydraulicSystemsData, I use type hints to specify what the expected argument and return types are.